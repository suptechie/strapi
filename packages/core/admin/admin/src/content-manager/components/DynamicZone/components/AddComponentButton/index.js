/**
 *
 * AddComponentButton
 *
 */

import React from 'react';
import PropTypes from 'prop-types';
import { useIntl } from 'react-intl';
import styled from 'styled-components';
import AddIcon from '@strapi/icons/AddIconCircle';
import { BaseButton } from '@strapi/parts/BaseButton';
import { Box } from '@strapi/parts/Box';
import { Row } from '@strapi/parts/Row';
import { Text, ButtonText, P } from '@strapi/parts/Text';
import { getTrad } from '../../../../utils';

const StyledAddIcon = styled(AddIcon)`
  transform: ${({ $isopen }) => ($isopen ? 'rotate(45deg)' : 'rotate(0deg)')};
  > circle {
    fill: ${({ theme }) => theme.colors.neutral150};
  }
  > path {
    fill: ${({ theme }) => theme.colors.neutral600};
  }
`;

const StyledButton = styled(BaseButton)`
  border-radius: 26px;
  background: ${({ theme }) => theme.colors.neutral0};
  padding: ${({ theme }) => theme.spaces[3]};
  border: ${({ hasError, theme }) => (hasError ? `1px solid ${theme.colors.danger600}` : '0')};
  box-shadow: ${({ theme }) => theme.shadows.filterShadow};

  svg {
    height: ${({ theme }) => theme.spaces[6]};
    width: ${({ theme }) => theme.spaces[6]};
    > path {
      fill: ${({ theme }) => theme.colors.neutral600};
    }
  }
  &:hover {
    color: ${({ theme }) => theme.colors.primary600} !important;
    ${Text} {
      color: ${({ theme }) => theme.colors.primary600} !important;
    }

    ${StyledAddIcon} {
      > circle {
        fill: ${({ theme }) => theme.colors.primary600};
      }
      > path {
        fill: ${({ theme }) => theme.colors.neutral100};
      }
    }
  }
  &:active {
    ${Text} {
      color: ${({ theme }) => theme.colors.primary600};
    }
    ${StyledAddIcon} {
      > circle {
        fill: ${({ theme }) => theme.colors.primary600};
      }
      > path {
        fill: ${({ theme }) => theme.colors.neutral100};
      }
    }
  }
`;

const BoxFullHeight = styled(Box)`
  height: 100%;
`;

const FieldError = ({ children, name }) => (
  <Box paddingTop={1}>
    <P
      small
      id={`${name}-error`}
      textColor="danger600"
      data-strapi-field-error
      style={{ textAlign: 'center' }}
    >
      {children}
    </P>
  </Box>
);

FieldError.propTypes = {
  children: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
};

const AddComponentButton = ({
  hasError,
  hasMaxError,
  hasMinError,
  isDisabled,
  isOpen,
  label,
  missingComponentNumber,
  name,
  onClick,
}) => {
  const { formatMessage } = useIntl();
  const addLabel = formatMessage(
    {
      id: getTrad('components.DynamicZone.add-component'),
      defaultMessage: 'Add a component to {componentName}',
    },
    { componentName: label || name }
  );
  const closeLabel = formatMessage({ id: 'app.utils.close-label', defaultMessage: 'Close' });

  return (
    <>
      <Row justifyContent="center">
        <Box
          paddingTop={6}
          paddingBottom={6}
          style={{ cursor: isDisabled ? 'not-allowed' : 'pointer' }}
        >
          <StyledButton type="button" onClick={onClick} disabled={isDisabled} hasError={hasError}>
            <Row>
              <BoxFullHeight aria-hidden paddingRight={2}>
                <StyledAddIcon $isopen={isOpen} />
              </BoxFullHeight>
              <ButtonText textColor="neutral500" small>
                {isOpen ? closeLabel : addLabel}
              </ButtonText>
            </Row>
          </StyledButton>

          {hasMaxError && !isOpen && (
            <FieldError name={name}>
              {formatMessage({ id: 'components.Input.error.validation.max' })}
            </FieldError>
          )}
          {hasMinError && !isOpen && (
            <FieldError name={name}>
              {formatMessage(
                {
                  id: getTrad(`components.DynamicZone.missing-components`),
                  defaultMessage:
                    'There {number, plural, =0 {are # missing components} one {is # missing component} other {are # missing components}}',
                },
                { number: missingComponentNumber }
              )}
            </FieldError>
          )}
        </Box>
      </Row>
    </>
  );
};

AddComponentButton.defaultProps = {
  label: '',
  missingComponentNumber: 0,
};

AddComponentButton.propTypes = {
  label: PropTypes.string,
  hasError: PropTypes.bool.isRequired,
  hasMaxError: PropTypes.bool.isRequired,
  hasMinError: PropTypes.bool.isRequired,
  isDisabled: PropTypes.bool.isRequired,
  isOpen: PropTypes.bool.isRequired,
  missingComponentNumber: PropTypes.number,
  name: PropTypes.string.isRequired,
  onClick: PropTypes.func.isRequired,
};

export default AddComponentButton;
