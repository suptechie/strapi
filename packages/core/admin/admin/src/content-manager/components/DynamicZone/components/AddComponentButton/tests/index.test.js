/**
 *
 * Tests for AddComponentButton
 *
 */

import React from 'react';
import { render } from '@testing-library/react';
import { ThemeProvider, lightTheme } from '@strapi/parts';
import { IntlProvider } from 'react-intl';
import AddComponentButton from '../index';

describe('<AddComponentButton />', () => {
  it('renders and matches the snapshot', () => {
    const {
      container: { firstChild },
    } = render(
      <ThemeProvider theme={lightTheme}>
        <IntlProvider locale="en" messages={{}} defaultLocale="en">
          <AddComponentButton label="test" name="name" onClick={jest.fn()} />
        </IntlProvider>
      </ThemeProvider>
    );

    expect(firstChild).toMatchInlineSnapshot(`
      .c1 {
        display: -webkit-box;
        display: -webkit-flex;
        display: -ms-flexbox;
        display: flex;
        cursor: pointer;
        padding: 8px;
        border-radius: 4px;
        background: #ffffff;
        border: 1px solid #dcdce4;
      }

      .c1 svg {
        height: 12px;
        width: 12px;
      }

      .c1 svg > g,
      .c1 svg path {
        fill: #ffffff;
      }

      .c1[aria-disabled='true'] {
        pointer-events: none;
      }

      .c4 {
        padding-right: 8px;
      }

      .c0 {
        display: -webkit-box;
        display: -webkit-flex;
        display: -ms-flexbox;
        display: flex;
        -webkit-flex-direction: row;
        -ms-flex-direction: row;
        flex-direction: row;
        -webkit-box-pack: center;
        -webkit-justify-content: center;
        -ms-flex-pack: center;
        justify-content: center;
        -webkit-align-items: center;
        -webkit-box-align: center;
        -ms-flex-align: center;
        align-items: center;
      }

      .c3 {
        display: -webkit-box;
        display: -webkit-flex;
        display: -ms-flexbox;
        display: flex;
        -webkit-flex-direction: row;
        -ms-flex-direction: row;
        flex-direction: row;
        -webkit-align-items: center;
        -webkit-box-align: center;
        -ms-flex-align: center;
        align-items: center;
      }

      .c9 {
        font-weight: 400;
        font-size: 0.75rem;
        line-height: 1.33;
        color: #8e8ea9;
      }

      .c10 {
        font-weight: 600;
        line-height: 1.14;
      }

      .c7 > circle {
        fill: #eaeaef;
      }

      .c7 > path {
        fill: #666687;
      }

      .c2 {
        border-radius: 26px;
        background: #ffffff;
        padding: 12px;
        border: 0;
      }

      .c2:hover {
        color: #4945ff !important;
      }

      .c2:hover .c8 {
        color: #4945ff !important;
      }

      .c2:hover .c6 > circle {
        fill: #4945ff;
      }

      .c2:hover .c6 > path {
        fill: #f6f6f9;
      }

      .c2:active .c8 {
        color: #4945ff;
      }

      .c2:active .c6 > circle {
        fill: #4945ff;
      }

      .c2:active .c6 > path {
        fill: #f6f6f9;
      }

      .c2 svg {
        height: 24px;
        width: 24px;
      }

      .c5 {
        height: 100%;
      }

      <div
        class="c0"
      >
        <button
          aria-disabled="false"
          class="c1 c2"
          type="button"
        >
          <div
            class="c3"
          >
            <div
              aria-hidden="true"
              class="c4 c5"
            >
              <svg
                class="c6 c7"
                fill="none"
                height="1em"
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <circle
                  cx="12"
                  cy="12"
                  fill="#212134"
                  r="12"
                />
                <path
                  d="M17 12.569c0 .124-.1.224-.225.224h-3.981v3.982c0 .124-.101.225-.226.225h-1.136a.225.225 0 01-.226-.225v-3.981H7.226A.225.225 0 017 12.567v-1.136c0-.125.1-.226.225-.226h3.982V7.226c0-.124.1-.225.224-.225h1.138c.124 0 .224.1.224.225v3.982h3.982c.124 0 .225.1.225.224v1.138z"
                  fill="#F6F6F9"
                />
              </svg>
            </div>
            <span
              class="c8 c9 c10"
            >
              Add a component to test
            </span>
          </div>
        </button>
      </div>
    `);
  });

  it('displays the name of the dz when the label is empty', () => {
    const {
      container: { firstChild },
    } = render(
      <ThemeProvider theme={lightTheme}>
        <IntlProvider locale="en" messages={{}} defaultLocale="en">
          <AddComponentButton label="" name="name" onClick={jest.fn()} />
        </IntlProvider>
      </ThemeProvider>
    );

    expect(firstChild).toMatchInlineSnapshot(`
      .c1 {
        display: -webkit-box;
        display: -webkit-flex;
        display: -ms-flexbox;
        display: flex;
        cursor: pointer;
        padding: 8px;
        border-radius: 4px;
        background: #ffffff;
        border: 1px solid #dcdce4;
      }

      .c1 svg {
        height: 12px;
        width: 12px;
      }

      .c1 svg > g,
      .c1 svg path {
        fill: #ffffff;
      }

      .c1[aria-disabled='true'] {
        pointer-events: none;
      }

      .c4 {
        padding-right: 8px;
      }

      .c0 {
        display: -webkit-box;
        display: -webkit-flex;
        display: -ms-flexbox;
        display: flex;
        -webkit-flex-direction: row;
        -ms-flex-direction: row;
        flex-direction: row;
        -webkit-box-pack: center;
        -webkit-justify-content: center;
        -ms-flex-pack: center;
        justify-content: center;
        -webkit-align-items: center;
        -webkit-box-align: center;
        -ms-flex-align: center;
        align-items: center;
      }

      .c3 {
        display: -webkit-box;
        display: -webkit-flex;
        display: -ms-flexbox;
        display: flex;
        -webkit-flex-direction: row;
        -ms-flex-direction: row;
        flex-direction: row;
        -webkit-align-items: center;
        -webkit-box-align: center;
        -ms-flex-align: center;
        align-items: center;
      }

      .c9 {
        font-weight: 400;
        font-size: 0.75rem;
        line-height: 1.33;
        color: #8e8ea9;
      }

      .c10 {
        font-weight: 600;
        line-height: 1.14;
      }

      .c7 > circle {
        fill: #eaeaef;
      }

      .c7 > path {
        fill: #666687;
      }

      .c2 {
        border-radius: 26px;
        background: #ffffff;
        padding: 12px;
        border: 0;
      }

      .c2:hover {
        color: #4945ff !important;
      }

      .c2:hover .c8 {
        color: #4945ff !important;
      }

      .c2:hover .c6 > circle {
        fill: #4945ff;
      }

      .c2:hover .c6 > path {
        fill: #f6f6f9;
      }

      .c2:active .c8 {
        color: #4945ff;
      }

      .c2:active .c6 > circle {
        fill: #4945ff;
      }

      .c2:active .c6 > path {
        fill: #f6f6f9;
      }

      .c2 svg {
        height: 24px;
        width: 24px;
      }

      .c5 {
        height: 100%;
      }

      <div
        class="c0"
      >
        <button
          aria-disabled="false"
          class="c1 c2"
          type="button"
        >
          <div
            class="c3"
          >
            <div
              aria-hidden="true"
              class="c4 c5"
            >
              <svg
                class="c6 c7"
                fill="none"
                height="1em"
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <circle
                  cx="12"
                  cy="12"
                  fill="#212134"
                  r="12"
                />
                <path
                  d="M17 12.569c0 .124-.1.224-.225.224h-3.981v3.982c0 .124-.101.225-.226.225h-1.136a.225.225 0 01-.226-.225v-3.981H7.226A.225.225 0 017 12.567v-1.136c0-.125.1-.226.225-.226h3.982V7.226c0-.124.1-.225.224-.225h1.138c.124 0 .224.1.224.225v3.982h3.982c.124 0 .225.1.225.224v1.138z"
                  fill="#F6F6F9"
                />
              </svg>
            </div>
            <span
              class="c8 c9 c10"
            >
              Add a component to name
            </span>
          </div>
        </button>
      </div>
    `);
  });
});
