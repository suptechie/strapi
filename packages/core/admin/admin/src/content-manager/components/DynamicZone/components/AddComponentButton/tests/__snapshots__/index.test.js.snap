// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AddComponentButton /> displays the name of the dz when the label is empty 1`] = `
.c12 {
  border: 0;
  -webkit-clip: rect(0 0 0 0);
  clip: rect(0 0 0 0);
  height: 1px;
  margin: -1px;
  overflow: hidden;
  padding: 0;
  position: absolute;
  width: 1px;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  cursor: pointer;
  padding: 8px;
  border-radius: 4px;
  background: #ffffff;
  border: 1px solid #dcdce4;
}

.c2 svg {
  height: 12px;
  width: 12px;
}

.c2 svg > g,
.c2 svg path {
  fill: #ffffff;
}

.c2[aria-disabled='true'] {
  pointer-events: none;
}

.c1 {
  padding-top: 24px;
  padding-bottom: 24px;
}

.c5 {
  padding-right: 8px;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c10 {
  font-weight: 400;
  font-size: 0.75rem;
  line-height: 1.33;
  color: #8e8ea9;
}

.c11 {
  font-weight: 600;
  line-height: 1.14;
}

.c8 {
  -webkit-transform: rotate(0deg);
  -ms-transform: rotate(0deg);
  transform: rotate(0deg);
}

.c8 > circle {
  fill: #eaeaef;
}

.c8 > path {
  fill: #666687;
}

.c3 {
  border-radius: 26px;
  background: #ffffff;
  padding: 12px;
  border: 0;
  box-shadow: 0px 1px 4px rgba(33,33,52,0.1);
}

.c3 svg {
  height: 24px;
  width: 24px;
}

.c3 svg > path {
  fill: #666687;
}

.c3:hover {
  color: #4945ff !important;
}

.c3:hover .c9 {
  color: #4945ff !important;
}

.c3:hover .c7 > circle {
  fill: #4945ff;
}

.c3:hover .c7 > path {
  fill: #f6f6f9;
}

.c3:active .c9 {
  color: #4945ff;
}

.c3:active .c7 > circle {
  fill: #4945ff;
}

.c3:active .c7 > path {
  fill: #f6f6f9;
}

.c6 {
  height: 100%;
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
      style="cursor: pointer;"
    >
      <button
        aria-disabled="false"
        class="c2 c3"
        type="button"
      >
        <div
          class="c4"
        >
          <div
            aria-hidden="true"
            class="c5 c6"
          >
            <svg
              class="c7 c8"
              fill="none"
              height="1em"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <circle
                cx="12"
                cy="12"
                fill="#212134"
                r="12"
              />
              <path
                d="M17 12.569c0 .124-.1.224-.225.224h-3.981v3.982c0 .124-.101.225-.226.225h-1.136a.225.225 0 01-.226-.225v-3.981H7.226A.225.225 0 017 12.567v-1.136c0-.125.1-.226.225-.226h3.982V7.226c0-.124.1-.225.224-.225h1.138c.124 0 .224.1.224.225v3.982h3.982c.124 0 .225.1.225.224v1.138z"
                fill="#F6F6F9"
              />
            </svg>
          </div>
          <span
            class="c9 c10 c11"
          >
            Add a component to name
          </span>
        </div>
      </button>
    </div>
  </div>
  <div
    class="c12"
  >
    <p
      aria-live="polite"
      aria-relevant="all"
      id="live-region-log"
      role="log"
    />
    <p
      aria-live="polite"
      aria-relevant="all"
      id="live-region-status"
      role="status"
    />
    <p
      aria-live="assertive"
      aria-relevant="all"
      id="live-region-alert"
      role="alert"
    />
  </div>
</div>
`;

exports[`<AddComponentButton /> renders and matches the snapshot 1`] = `
.c12 {
  border: 0;
  -webkit-clip: rect(0 0 0 0);
  clip: rect(0 0 0 0);
  height: 1px;
  margin: -1px;
  overflow: hidden;
  padding: 0;
  position: absolute;
  width: 1px;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  cursor: pointer;
  padding: 8px;
  border-radius: 4px;
  background: #ffffff;
  border: 1px solid #dcdce4;
}

.c2 svg {
  height: 12px;
  width: 12px;
}

.c2 svg > g,
.c2 svg path {
  fill: #ffffff;
}

.c2[aria-disabled='true'] {
  pointer-events: none;
}

.c1 {
  padding-top: 24px;
  padding-bottom: 24px;
}

.c5 {
  padding-right: 8px;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c10 {
  font-weight: 400;
  font-size: 0.75rem;
  line-height: 1.33;
  color: #8e8ea9;
}

.c11 {
  font-weight: 600;
  line-height: 1.14;
}

.c8 {
  -webkit-transform: rotate(0deg);
  -ms-transform: rotate(0deg);
  transform: rotate(0deg);
}

.c8 > circle {
  fill: #eaeaef;
}

.c8 > path {
  fill: #666687;
}

.c3 {
  border-radius: 26px;
  background: #ffffff;
  padding: 12px;
  border: 0;
  box-shadow: 0px 1px 4px rgba(33,33,52,0.1);
}

.c3 svg {
  height: 24px;
  width: 24px;
}

.c3 svg > path {
  fill: #666687;
}

.c3:hover {
  color: #4945ff !important;
}

.c3:hover .c9 {
  color: #4945ff !important;
}

.c3:hover .c7 > circle {
  fill: #4945ff;
}

.c3:hover .c7 > path {
  fill: #f6f6f9;
}

.c3:active .c9 {
  color: #4945ff;
}

.c3:active .c7 > circle {
  fill: #4945ff;
}

.c3:active .c7 > path {
  fill: #f6f6f9;
}

.c6 {
  height: 100%;
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
      style="cursor: pointer;"
    >
      <button
        aria-disabled="false"
        class="c2 c3"
        type="button"
      >
        <div
          class="c4"
        >
          <div
            aria-hidden="true"
            class="c5 c6"
          >
            <svg
              class="c7 c8"
              fill="none"
              height="1em"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <circle
                cx="12"
                cy="12"
                fill="#212134"
                r="12"
              />
              <path
                d="M17 12.569c0 .124-.1.224-.225.224h-3.981v3.982c0 .124-.101.225-.226.225h-1.136a.225.225 0 01-.226-.225v-3.981H7.226A.225.225 0 017 12.567v-1.136c0-.125.1-.226.225-.226h3.982V7.226c0-.124.1-.225.224-.225h1.138c.124 0 .224.1.224.225v3.982h3.982c.124 0 .225.1.225.224v1.138z"
                fill="#F6F6F9"
              />
            </svg>
          </div>
          <span
            class="c9 c10 c11"
          >
            Add a component to test
          </span>
        </div>
      </button>
    </div>
  </div>
  <div
    class="c12"
  >
    <p
      aria-live="polite"
      aria-relevant="all"
      id="live-region-log"
      role="log"
    />
    <p
      aria-live="polite"
      aria-relevant="all"
      id="live-region-status"
      role="status"
    />
    <p
      aria-live="assertive"
      aria-relevant="all"
      id="live-region-alert"
      role="alert"
    />
  </div>
</div>
`;

exports[`<AddComponentButton /> renders and matches the snapshot when the isOpen prop is truthy 1`] = `
.c12 {
  border: 0;
  -webkit-clip: rect(0 0 0 0);
  clip: rect(0 0 0 0);
  height: 1px;
  margin: -1px;
  overflow: hidden;
  padding: 0;
  position: absolute;
  width: 1px;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  cursor: pointer;
  padding: 8px;
  border-radius: 4px;
  background: #ffffff;
  border: 1px solid #dcdce4;
}

.c2 svg {
  height: 12px;
  width: 12px;
}

.c2 svg > g,
.c2 svg path {
  fill: #ffffff;
}

.c2[aria-disabled='true'] {
  pointer-events: none;
}

.c1 {
  padding-top: 24px;
  padding-bottom: 24px;
}

.c5 {
  padding-right: 8px;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c10 {
  font-weight: 400;
  font-size: 0.75rem;
  line-height: 1.33;
  color: #8e8ea9;
}

.c11 {
  font-weight: 600;
  line-height: 1.14;
}

.c8 {
  -webkit-transform: rotate(45deg);
  -ms-transform: rotate(45deg);
  transform: rotate(45deg);
}

.c8 > circle {
  fill: #eaeaef;
}

.c8 > path {
  fill: #666687;
}

.c3 {
  border-radius: 26px;
  background: #ffffff;
  padding: 12px;
  border: 0;
  box-shadow: 0px 1px 4px rgba(33,33,52,0.1);
}

.c3 svg {
  height: 24px;
  width: 24px;
}

.c3 svg > path {
  fill: #666687;
}

.c3:hover {
  color: #4945ff !important;
}

.c3:hover .c9 {
  color: #4945ff !important;
}

.c3:hover .c7 > circle {
  fill: #4945ff;
}

.c3:hover .c7 > path {
  fill: #f6f6f9;
}

.c3:active .c9 {
  color: #4945ff;
}

.c3:active .c7 > circle {
  fill: #4945ff;
}

.c3:active .c7 > path {
  fill: #f6f6f9;
}

.c6 {
  height: 100%;
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
      style="cursor: pointer;"
    >
      <button
        aria-disabled="false"
        class="c2 c3"
        type="button"
      >
        <div
          class="c4"
        >
          <div
            aria-hidden="true"
            class="c5 c6"
          >
            <svg
              class="c7 c8"
              fill="none"
              height="1em"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <circle
                cx="12"
                cy="12"
                fill="#212134"
                r="12"
              />
              <path
                d="M17 12.569c0 .124-.1.224-.225.224h-3.981v3.982c0 .124-.101.225-.226.225h-1.136a.225.225 0 01-.226-.225v-3.981H7.226A.225.225 0 017 12.567v-1.136c0-.125.1-.226.225-.226h3.982V7.226c0-.124.1-.225.224-.225h1.138c.124 0 .224.1.224.225v3.982h3.982c.124 0 .225.1.225.224v1.138z"
                fill="#F6F6F9"
              />
            </svg>
          </div>
          <span
            class="c9 c10 c11"
          >
            Close
          </span>
        </div>
      </button>
    </div>
  </div>
  <div
    class="c12"
  >
    <p
      aria-live="polite"
      aria-relevant="all"
      id="live-region-log"
      role="log"
    />
    <p
      aria-live="polite"
      aria-relevant="all"
      id="live-region-status"
      role="status"
    />
    <p
      aria-live="assertive"
      aria-relevant="all"
      id="live-region-alert"
      role="alert"
    />
  </div>
</div>
`;
